test/e2e/e2e.py

```diff
@@ -0,0 +1,438 @@
+    f = tempfile.NamedTemporaryFile()
+    t = tarfile.open(mode="w", fileobj=f)
+
+    abs_path = os.path.abspath(path)
+    t.add(abs_path, arcname=os.path.basename(path), recursive=False)
+
+    t.close()
+    f.seek(0)
+    return f
+
+
+def async_print_container_log(container):
+    x = threading.Thread(target=print_container_log, args=(container,), daemon=True)
+    x.start()
+
+
+def print_container_log(container):
+    for log in container.logs(stream=True):
+        print(f"{container.name}>{log}")
+
+
+def current_time():
+    # return datetime.now().strftime("%Y%m%d-%H%M%S-%f")
+    return time.time()
+
+
+def random_string(length=8):
+    return "".join(
+        [random.choice(string.ascii_letters + string.digits) for n in range(length)]
+    )
+
+
+def random_id():
+    return "{:0>3}".format(random.randrange(999))
+
+
+def redis_client_call(redis_client, command, args):
+    # cmd_func = getattr(redis_client, command)
+    # return cmd_func(**args)
+
+    # expanded_args = (**args)
+    args_tupled = args
+    if not isinstance(args, tuple):
+        args_tupled = (args,)
+    # print(f"### COMMAND: {command} ARGS: {args_tupled} LEN_ARGS: {len(args_tupled)}")
+
+    return redis_client.execute_command(command, *args_tupled)
+
+
+def redis_client_call_inbulk(redis_client, list_command):
+    for cmd_n_args in list_command:
+        for cmd, args in cmd_n_args.items():
+            # print(">> CMD:{} ARGS:{}".format(cmd, args))
+            redis_client_call(redis_client, cmd, args)
+
+
+def run_container(client, name, image, command, network, volumes, ports=None):
+    print(f"Creating container: {name}")
+    container = client.containers.run(
+        name=name,
+        image=image,
+        detach=True,
+        command=command,
+        network=network,
+        volumes=volumes,
+        ports=ports,
+    )
+    # pprint(container.attrs)
+    return container
+
+
+def run_test(client, api_client, remiro_image, rdb_tools_image, e2e_id, test_case):
+    # tc_id = "tid{}".format(random_id())
+    tc_id = test_case["id"]
+    tc_test = test_case["test"]
+
+    print("=== Preparing a test ===")
+    print(f"tc_id: {tc_id} test_name: {test_case['name']}")
+
+    print("Creating 'e2e-test-network' ...")
+    e2e_test_network = client.networks.create(f"e2e-test-network-{e2e_id}-{tc_id}")
+    pprint(e2e_test_network)
+
+    print("Creating 'e2e-test-volume' ...")
+    e2e_test_volume = client.volumes.create(f"e2e-test-volume-{e2e_id}-{tc_id}")
+    pprint(e2e_test_volume)
+
+    # === setup volume container: intermediary container to copy files from host to volume ===
+    client.images.pull("hello-world:latest")
+    e2e_test_volume_container = client.containers.create(
+        name=f"e2e-T-volume-container-{e2e_id}-{tc_id}",
+        image="hello-world",
+        volumes={e2e_test_volume.name: DEFAULT_BIND_VOLUME},
+    )
+    # ===
+    print("Creating containers ...")
+
+    redis_src_container = run_container(
```

Comment:
Can we wrap the run_container used by the backing Redis into another function? The only parameters that differs are the` name` `dbfilename`, it would help the `run_test` readibility, which is already quite long as it is.

Reply:
Done. 5bde833